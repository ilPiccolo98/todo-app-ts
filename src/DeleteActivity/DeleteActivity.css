.delete-submit {
  margin-top: 18px;
}

.error-id {
  position: relative;
  left: 35%;
  margin-top: 10px;
}

.message-description {
  position: relative;
  left: 27%;
  margin-top: 10px;
}
