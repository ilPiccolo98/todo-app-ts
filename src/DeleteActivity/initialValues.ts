export type DeleteActivityValues = {
  id: string;
};

const initialValue: DeleteActivityValues = {
  id: "",
};

export default initialValue;
