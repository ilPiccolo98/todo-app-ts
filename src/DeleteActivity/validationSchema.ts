import * as yup from "yup";

const validationSchema = yup.object().shape({
  id: yup.number().required("Id is required"),
});

export default validationSchema;
