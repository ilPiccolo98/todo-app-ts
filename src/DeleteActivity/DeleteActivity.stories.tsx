import React from "react";
import { Meta, Story } from "@storybook/react";
import { DeleteActivity, DeleteActivityProps } from "./DeleteActivity";
import store from "../activities/activitiesStore";
import { Provider } from "react-redux";

const meta: Meta = {
  title: "DeleteActivity",
  component: DeleteActivity,
  argTypes: {},
};

export default meta;
const Template: Story<DeleteActivityProps> = (args) => (
  <Provider store={store}>
    <DeleteActivity {...args} />
  </Provider>
);

export const Default = Template.bind({});
Default.args = {};
