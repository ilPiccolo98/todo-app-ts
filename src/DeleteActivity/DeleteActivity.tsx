import React, { useEffect } from "react";
import {
  deleteActivity,
  DeleteState,
  deleteStateSelector,
  fetchAllActivities,
  resetDeleteState,
} from "../activities/activitiesSlice";
import { useDispatch, useSelector } from "react-redux";
import initialValues from "./initialValues";
import validationSchema from "./validationSchema";
import Form from "../components/Form/Form";
import Submit from "../components/Submit/Submit";
import TextField from "../components/TextField/TextField";
import "./DeleteActivity.css";
import { useFormik } from "formik";
import MessageError from "../components/MessageError/MessageError";
import { RootState } from "../activities/activitiesStore";
import Message from "../components/Message/Message";

export interface DeleteActivityProps {}

export const DeleteActivity: React.FC<DeleteActivityProps> = (): JSX.Element => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(resetDeleteState());
  }, [dispatch]);

  const deleteState: DeleteState = useSelector<RootState, DeleteState>(
    deleteStateSelector
  );

  useEffect(() => {
    if (deleteState === DeleteState.Deleted) {
      dispatch(fetchAllActivities());
    }
  }, [dispatch, deleteState]);

  const formik = useFormik({
    initialValues,
    validationSchema,
    onSubmit: (values) => {
      dispatch(deleteActivity(+values.id));
    },
  });

  const getMessage = (): JSX.Element => {
    if (deleteState === DeleteState.Deleted) {
      return (
        <Message size="large" variant="green">
          Activity Deleted
        </Message>
      );
    } else if (deleteState === DeleteState.Deleting) {
      return (
        <Message size="large" variant="blue">
          Deleting Activity
        </Message>
      );
    } else if (deleteState === DeleteState.NotDeleted) {
      return <MessageError size="small">Activity not deleted!</MessageError>;
    }
    return <></>;
  };

  return (
    <Form
      title="Delete Activity"
      variant="primary"
      className="form-delete"
      onSubmit={formik.handleSubmit}
    >
      <TextField
        id="id"
        name="id"
        variant="secondary"
        type="numbers"
        labelText="Id"
        placeholder="Insert Id"
        onChange={formik.handleChange}
        value={formik.values.id}
        datatestid="id-deleteActivity"
      />
      <Submit
        datatestid="submit-deleteActivity"
        className="delete-submit"
        variant="primary"
      >
        Delete
      </Submit>
      <div className="error-id">
        <MessageError size="medium">{formik.errors.id}</MessageError>
      </div>
      <div className="message-description">{getMessage()}</div>
    </Form>
  );
};

export default DeleteActivity;
