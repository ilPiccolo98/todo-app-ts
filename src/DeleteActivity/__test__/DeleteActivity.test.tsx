import React from "react";
import {
  fireEvent,
  render,
  RenderResult,
  waitFor,
} from "@testing-library/react";
import DeleteActivity from "../DeleteActivity";
import { Provider } from "react-redux";
import "@testing-library/jest-dom/extend-expect";
import * as actions from "../../activities/activitiesSlice";
import store from "../../activities/activitiesStore";

const renderDeleteActivity = () => {
  return render(
    <Provider store={store}>
      <DeleteActivity />
    </Provider>
  );
};

const getDeleteActivityComponents = (
  component: RenderResult<
    typeof import("@testing-library/dom/types/queries"),
    HTMLElement
  >
) => {
  const idTextField = component.getByTestId("id-deleteActivity").childNodes[0]
    .childNodes[1] as HTMLElement;
  const submit = component.getByTestId(
    "submit-deleteActivity"
  ) as HTMLInputElement;
  return { idTextField, submit };
};

describe("testing DeleteActivity component", () => {
  const deleteStateSelectorSpy = jest.spyOn(actions, "deleteStateSelector");
  const fetchAllActivitiesSpy = jest.spyOn(actions, "fetchAllActivities");
  const activitiesSelectorSpy = jest.spyOn(actions, "activitiesSelector");
  const deleteActivitySpy = jest.spyOn(actions, "deleteActivity");
  beforeEach(() => {
    activitiesSelectorSpy.mockReset().mockReturnValue([]);
    deleteActivitySpy.mockClear();
    deleteStateSelectorSpy
      .mockReset()
      .mockReturnValue(actions.DeleteState.Null);
    fetchAllActivitiesSpy.mockClear();
  });

  it("should call the deleteActivity actions with { id: 1 }", async () => {
    activitiesSelectorSpy.mockReset().mockReturnValue([
      {
        id: 1,
        name: "activity",
        description: "activity",
        status: true,
      },
    ]);
    const component = renderDeleteActivity();
    const { idTextField, submit } = getDeleteActivityComponents(component);
    await waitFor(() => {
      fireEvent.change(idTextField, {
        target: {
          value: "1",
        },
      });
    });
    await waitFor(() => {
      fireEvent.click(submit);
    });
    expect(deleteActivitySpy).toHaveBeenCalledWith(1);
    expect(component).toMatchSnapshot();
  });

  it("shouldn't call the deleteActivity actions when id field is blank", async () => {
    activitiesSelectorSpy.mockReset().mockReturnValue([
      {
        id: 1,
        name: "activity",
        description: "activity",
        status: true,
      },
    ]);
    const component = renderDeleteActivity();
    const { submit } = getDeleteActivityComponents(component);
    await waitFor(() => {
      fireEvent.click(submit);
    });
    expect(deleteActivitySpy).not.toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

  it("should call the fetchAllActivities action once when deleteState === DeleteState.Deleted", () => {
    deleteStateSelectorSpy
      .mockReset()
      .mockReturnValue(actions.DeleteState.Deleted);
    const component = renderDeleteActivity();
    expect(fetchAllActivitiesSpy).toHaveBeenCalledTimes(1);
    expect(component).toMatchSnapshot();
  });

  it("should show the message Activity Deleted when deleteState === DeleteState.Deleted", () => {
    deleteStateSelectorSpy
      .mockReset()
      .mockReturnValue(actions.DeleteState.Deleted);
    const component = renderDeleteActivity();
    const message = component.getByText(/Activity Deleted/i);
    expect(message.textContent).toBe("Activity Deleted");
    expect(component).toMatchSnapshot();
  });

  it("should show the message Activity Deleted when deleteState === DeleteState.Deleting", () => {
    deleteStateSelectorSpy
      .mockReset()
      .mockReturnValue(actions.DeleteState.Deleting);
    const component = renderDeleteActivity();
    const message = component.getByText(/Deleting Activity/i);
    expect(message.textContent).toBe("Deleting Activity");
    expect(component).toMatchSnapshot();
  });

  it("should show the message Activity Deleted when deleteState === DeleteState.NotDeleted", () => {
    deleteStateSelectorSpy
      .mockReset()
      .mockReturnValue(actions.DeleteState.NotDeleted);
    const component = renderDeleteActivity();
    const message = component.getByText(/Activity not deleted!/i);
    expect(message.textContent).toBe("Activity not deleted!");
    expect(component).toMatchSnapshot();
  });
});
