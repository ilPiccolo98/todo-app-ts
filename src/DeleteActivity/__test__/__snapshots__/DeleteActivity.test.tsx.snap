// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing DeleteActivity component should call the deleteActivity actions with { id: 1 } 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value="1"
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="message-description"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value="1"
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b />
          </span>
        </div>
        <div
          class="message-description"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing DeleteActivity component should call the fetchAllActivities action once when deleteState === DeleteState.Deleted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value=""
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="message-description"
          >
            <span
              class="undefined message-variant-green message-size-large"
              variant="green"
            >
              Activity Deleted
            </span>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value=""
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b />
          </span>
        </div>
        <div
          class="message-description"
        >
          <span
            class="undefined message-variant-green message-size-large"
            variant="green"
          >
            Activity Deleted
          </span>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing DeleteActivity component should show the message Activity Deleted when deleteState === DeleteState.Deleted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value=""
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="message-description"
          >
            <span
              class="undefined message-variant-green message-size-large"
              variant="green"
            >
              Activity Deleted
            </span>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value=""
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b />
          </span>
        </div>
        <div
          class="message-description"
        >
          <span
            class="undefined message-variant-green message-size-large"
            variant="green"
          >
            Activity Deleted
          </span>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing DeleteActivity component should show the message Activity Deleted when deleteState === DeleteState.Deleting 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value=""
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="message-description"
          >
            <span
              class="undefined message-variant-blue message-size-large"
              variant="blue"
            >
              Deleting Activity
            </span>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value=""
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b />
          </span>
        </div>
        <div
          class="message-description"
        >
          <span
            class="undefined message-variant-blue message-size-large"
            variant="blue"
          >
            Deleting Activity
          </span>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing DeleteActivity component should show the message Activity Deleted when deleteState === DeleteState.NotDeleted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value=""
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="message-description"
          >
            <span
              class="message-error undefined small"
            >
              <b>
                Activity not deleted!
              </b>
            </span>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value=""
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b />
          </span>
        </div>
        <div
          class="message-description"
        >
          <span
            class="message-error undefined small"
          >
            <b>
              Activity not deleted!
            </b>
          </span>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing DeleteActivity component shouldn't call the deleteActivity actions when id field is blank 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form form-delete"
      >
        <p>
          <b>
            Delete Activity
          </b>
        </p>
        <form
          class=""
        >
          <div
            class="secondary-container"
            data-testid="id-deleteActivity"
          >
            <label
              for="id"
            >
              <p
                class="secondary-label"
              >
                <b>
                  Id
                </b>
              </p>
              <input
                class="textField undefined"
                datatestid="id-deleteActivity"
                id="id"
                name="id"
                placeholder="Insert Id"
                type="numbers"
                value=""
                variant="secondary"
              />
            </label>
            <div
              class="error"
            >
              <span
                class="message-error undefined small"
              >
                <b />
              </span>
            </div>
          </div>
          <input
            class="submit primary-submit delete-submit"
            data-testid="submit-deleteActivity"
            datatestid="submit-deleteActivity"
            type="submit"
            value="Delete"
          />
          <div
            class="error-id"
          >
            <span
              class="message-error undefined medium"
            >
              <b>
                Id is required
              </b>
            </span>
          </div>
          <div
            class="message-description"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form form-delete"
    >
      <p>
        <b>
          Delete Activity
        </b>
      </p>
      <form
        class=""
      >
        <div
          class="secondary-container"
          data-testid="id-deleteActivity"
        >
          <label
            for="id"
          >
            <p
              class="secondary-label"
            >
              <b>
                Id
              </b>
            </p>
            <input
              class="textField undefined"
              datatestid="id-deleteActivity"
              id="id"
              name="id"
              placeholder="Insert Id"
              type="numbers"
              value=""
              variant="secondary"
            />
          </label>
          <div
            class="error"
          >
            <span
              class="message-error undefined small"
            >
              <b />
            </span>
          </div>
        </div>
        <input
          class="submit primary-submit delete-submit"
          data-testid="submit-deleteActivity"
          datatestid="submit-deleteActivity"
          type="submit"
          value="Delete"
        />
        <div
          class="error-id"
        >
          <span
            class="message-error undefined medium"
          >
            <b>
              Id is required
            </b>
          </span>
        </div>
        <div
          class="message-description"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
